java/core/src/java/org/apache/orc/util/BloomFilter.java

```diff
@@ -130,7 +125,7 @@ public void addString(String val) {
     if (val == null) {
       add(null);
     } else {
-      add(val.getBytes());
+      add(val.getBytes(Charset.defaultCharset()));
```

Comment:
No, this one is explicitly the default charset to be compatible. BloomFilterUtf8 is the fixed one.